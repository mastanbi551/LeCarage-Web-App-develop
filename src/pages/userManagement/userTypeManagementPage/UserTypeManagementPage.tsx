import React from 'react';
import {
  DynamicDataTable,
  PageHeader,
} from '../../../components';
import { BookingInfo } from '../../bookingManagement/BookingsManagementPage';
import { CarProps } from '../carManagementPage/CarManagementPage';
import { CustomerProps } from '../customerManagementPage/CustomerManagementPage';
import { userProps } from '../userManagementPage/UserManagementPage';
import './UserTypeManagementPage.scss';


export interface UserTypeProps {
  Description: string,
  Status: string,
  'User Type': string,
  'User Type ID': string
}

interface UserTypesInterfaceProps {
  handleAddUsertype: () => void;
  columns: { Header: string; accessor: string; }[];
  searchValue: string[],
  onViewHandler: (rowInfo: object) => void;
  onEditHandler: (rowInfo: object) => void;
  onDeleteHandler: (rowInfo: object) => void;
  userTypesInfo: [];
  filterValues: {mainMenu: string, subMenus: string[]}[];
  errorMessage: string;
  access: string;
  itemsPerPage: number;
  rowsPerPageValueChange: (value: string) => void;
}
const UserTypeManagementPage: React.FC<UserTypesInterfaceProps> = ({
  handleAddUsertype,
  columns,
  searchValue,
  onViewHandler,
  onEditHandler,
  onDeleteHandler,
  userTypesInfo,
  filterValues,
  errorMessage,
  access,
  itemsPerPage,
  rowsPerPageValueChange
}) => {
  return (
    <>
      <div className='user-types'>
        <div className='user-types__header-content'>
          <PageHeader
            text='User Types'
            name='user-types__header-content__heading'
            isButtonDisable={true}
            buttonName='Add New'
            buttonClassName='user-types__header-content__button greenButton btn btn-success'
            buttonOnclick={handleAddUsertype}
            access={access}
          />
        </div>

        <div className='user-types__table-content '>
          <div className='user-types__table'>
            <DynamicDataTable
              tableHeading=''
              isSearchIconEnable={true}
              isFilterIconEnable={true}
              columns={columns}
              data={userTypesInfo}
              searchValue={searchValue}
              filterValues={filterValues}
              onViewHandler={onViewHandler}
              onEditHandler={onEditHandler}
              onDeleteHandler={onDeleteHandler}
              defaultItemsPerPage={itemsPerPage}
              errorMessage={errorMessage}
              isEditIconEnable={true}
              isDeleteIconEnable={true}
              access={access}
              rowsPerPageValueChange={rowsPerPageValueChange}
            />
          </div>
        </div>
      </div>
    </>
  );
};

export default UserTypeManagementPage;