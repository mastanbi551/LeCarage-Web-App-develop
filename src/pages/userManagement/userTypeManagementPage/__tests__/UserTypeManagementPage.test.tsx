import React from 'react';
import { render, screen } from '@testing-library/react';
import UserTypeManagementPage from '../UserTypeManagementPage';

describe('UserTypeManagementPage', () => {
  const handleAddUsertypeMock = jest.fn();
  const onViewHandlerMock = jest.fn();
  const onEditHandlerMock = jest.fn();
  const onDeleteHandlerMock = jest.fn();

  const columns = [
    { Header: 'User Type ID', accessor: 'User Type ID' },
    { Header: 'User Type', accessor: 'User Type' },
    { Header: 'Description', accessor: 'Description' },
    { Header: 'Status', accessor: 'Status' },
  ];

  const userTypesInfo = [
    {
      'User Type ID': '1',
      'User Type': 'Admin',
      Description: 'Admin User Type',
      Status: 'Active',
    },
    {
      'User Type ID': '2',
      'User Type': 'User',
      Description: 'User User Type',
      Status: 'Inactive',
    },
  ];

  const searchValue = ['Admin'];
  const filterValues = [{ mainMenu: 'Filter', subMenus: ['Admin'] }];
  const errorMessage = 'Error message';

  it('should match the snapshot', () => {
    const { container } = render(
      <UserTypeManagementPage
        handleAddUsertype={handleAddUsertypeMock}
        columns={columns}
        searchValue={searchValue}
        onViewHandler={onViewHandlerMock}
        onEditHandler={onEditHandlerMock}
        onDeleteHandler={onDeleteHandlerMock}
        userTypesInfo={userTypesInfo}
        filterValues={filterValues}
        errorMessage={errorMessage}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should call handleAddUsertype when Add New button is clicked', () => {
    render(
      <UserTypeManagementPage
        handleAddUsertype={handleAddUsertypeMock}
        columns={columns}
        searchValue={searchValue}
        onViewHandler={onViewHandlerMock}
        onEditHandler={onEditHandlerMock}
        onDeleteHandler={onDeleteHandlerMock}
        userTypesInfo={userTypesInfo}
        filterValues={filterValues}
        errorMessage={errorMessage}
      />
    );

    const addButton = screen.getByRole('button', { name: 'Add New' });
    addButton.click();

    expect(handleAddUsertypeMock).toHaveBeenCalled();
  });

 

});