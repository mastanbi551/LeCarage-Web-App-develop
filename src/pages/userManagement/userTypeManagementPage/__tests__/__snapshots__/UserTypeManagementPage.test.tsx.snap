// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTypeManagementPage should match the snapshot 1`] = `
<div>
  <div
    class="user-types"
  >
    <div
      class="user-types__header-content"
    >
      <h1
        class="user-types__header-content__heading"
      >
        User Types
      </h1>
      <button
        class="user-types__header-content__button greenButton btn btn-success"
        type="button"
      >
        Add New
      </button>
    </div>
    <div
      class="user-types__table-content "
    >
      <div
        class="user-types__table"
      >
        <div
          class="text-align-center norecord"
        >
          Error message
        </div>
      </div>
    </div>
  </div>
</div>
`;
